app/src/pages/addBidPage/addBid.jsx

```diff
@@ -0,0 +1,257 @@
+            <div className={styles.imagePreview}  style={{ backgroundImage : `url(${details.src})` }}></div>
+            <p>{details.name}</p>
+        </div>
+    )
+}
+
+
+export default function AddBid (props){
+    const postId = useParams().id;
+
+    const [postData, setPostData] = useState({})
+    const imageBtns = [0, 1, 2]
+    const [imagePortfolios, setImagePortfolios] = useState([[], [], []])
+
+    const [currentIndex, setCurrentIndex] = useState(0)
+    // put in separate state to force re-renders
+    const [currentImages, setCurrentImages]  = useState([])
+
+    const [description, setDescription] = useState("")
+
+    const fileInputRef = useRef(null);
+    const descRef = useRef(null);
+
+    const reader = new FileReader()
+    const cookies = new Cookies(null, {path: '/'})
+    const userId = cookies.get('currentUser').id
+
+
+    const [finish, setFinish] = useState(false)
+    const [navHome, setNavHome] = useState(false)
+
+    // TODO: this is a re-fetch, find ways to optimize
+    useEffect(() => {
+        async function getData(id){
+            const data = await API.getPost(id);
+            setPostData(data)
+        }
+
+        getData(postId)
+    }, [])
+
+    useEffect(() => {
+        setCurrentImages(imagePortfolios[currentIndex])
+    }, [currentIndex, imagePortfolios[currentIndex]])
+
+
+    const handleUpload = (event) => {
+        // routine to preview portfolio image before adding it to big
+        // called using useRef and div click
+        const file = event.target.files[0]
+        let imageData = []
+
+        var result = ""
+        if (file) {
+            reader.onload = (e) => {
+                result = e.target.result
+                imageData = [result, file.name]
+
+                setImagePortfolios((prev) => {
+                    const temp = [...prev];
+                    temp[currentIndex] = [...new Set([...temp[currentIndex], [result, file.name]])]
+                    return temp;
+                })
+
+            }
+            reader.readAsDataURL(file)
+
+        }
+    }
+
+    const changeDescriptionNext = () => {
+        if (currentIndex < 2) {
+            setCurrentIndex((prev) => {
+                return prev+1}
+            )
+        }
+    }
+
+    const handlePost = async () => {
+        if (description){
+            const allImageData = []
+
+            for (let i in imagePortfolios){
+                const imageData = []
+
+                for (let image in imagePortfolios[i]){
+                    imageData.push({
+                        "name" : imagePortfolios[i][image][1],
+                        "blob" : imagePortfolios[i][image][0]
+                    })
+                }
+                allImageData.push(imageData)
+            }
+
+            await API.addBid(allImageData, description, postId, userId)
+        }
+
+        setNavHome(true)
+
+    }
+
+
+    return (
+        <div id={styles.addBid}>
```

Comment:
Thanks for the attention to detail! lots of off by one and special index handling in here